base/collection_view.go

```diff
@@ -202,19 +204,34 @@ func (c *Collection) executeViewQuery(ddoc, name string, params map[string]inter
 	return goCbViewResult.Raw(), nil
 }
 
+type gocbResultRaw interface {
+
+	// NextBytes returns the next row as bytes.
+	NextBytes() []byte
+
+	// Err returns any errors that have occurred on the stream
+	Err() error
+
+	// Close marks the results as closed, returning any errors that occurred during reading the results.
+	Close() error
+
+	// MetaData returns any meta-data that was available from this query as bytes.
+	MetaData() ([]byte, error)
+}
```

Comment:
This interface and the rawIterator doesn't look view-specific, looks like a common interface across all gocb row readers.

Could probably be moved into `collection_n1ql_common.go`, unless we want to rename it to make it view specific?